crates/starknet-server/tests/test_messaging.rs

```diff
@@ -0,0 +1,229 @@
+//! This module tests the messaging feature from the starknet-server perspective.
+//!
+//! The message contract `cairo_0_l1l2_contract` associated a balance to a user (ContractAddress)
+//! and contains the following entrypoints:
+//! * increase_balance -> increases the balance of a user (contract address) for the given amount.
+//! * get_balance -> gets the balance of a user.
+//! * withdraw -> withdraw from a user the given amount, sending the amount to a l2->l1 message.
+//! * deposit -> deposit the given amount from a l1->l2 message (l1 handler function).
+
+// must use `pub`: https://github.com/rust-lang/rust/issues/46379#issuecomment-548787629
+pub mod common;
+
+mod test_messaging {
+    use std::sync::Arc;
+
+    use hyper::{Body, StatusCode};
+    use serde_json::json;
+    use starknet_core::utils::exported_test_utils::dummy_cairo_l1l2_contract;
+    use starknet_rs_accounts::{
+        Account, Call, ConnectedAccount, ExecutionEncoding, SingleOwnerAccount,
+    };
+    use starknet_rs_contract::ContractFactory;
+    use starknet_rs_core::types::contract::legacy::LegacyContractClass;
+    use starknet_rs_core::types::{BlockId, BlockTag, FieldElement, FunctionCall};
+    use starknet_rs_core::utils::{
+        get_selector_from_name, get_udc_deployed_address, UdcUniqueness,
+    };
+    use starknet_rs_providers::jsonrpc::HttpTransport;
+    use starknet_rs_providers::{JsonRpcClient, Provider};
+    use starknet_rs_signers::LocalWallet;
+
+    use crate::common::constants::{CHAIN_ID, MESSAGING_L1_ALLOWED_CONTRACT};
+    use crate::common::devnet::BackgroundDevnet;
```

Comment:
Note that `crate::common::devnet` has been renamed to `crate::common::background_devnet`

Reply:
Thanks for the info, was adjusted accordingly.